server/src/services/UserService.ts

```diff
@@ -85,4 +112,44 @@ export class UserService {
       throw new Error(`Error fetching users from UserService.getUsers: ${e}`);
     }
   }
-}
\ No newline at end of file
+
+  public static async edit(user: EditUser, ctx: Context) {
+    try {
+      const currUser = await AuthUtil.verifyAndGetUser(ctx);
+
+      if (!currUser) {
+        throw new Error("Error retrieving logged in user");
+      }
+
+      const existingUser = await ctx.prisma.user.findUnique({
+        where: { email: user.email },
+      });
+
+      if (existingUser) {
+        throw new Error("email has already been taken");
```

Comment:
you found the current user, and then returned an error if it exists, wont this always be hit? Whats the point of this